src/cudnn/result.rs

```diff
@@ -0,0 +1,63 @@
+//! A thin wrapper around [sys] providing [Result]s with [CudnnError].
+
+use core::mem::MaybeUninit;
+
+use crate::device::CudaDevice;
+
+use super::sys::*;
+
+pub type CudnnResult<T> = Result<T, CudnnError>;
+
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub struct CudnnError(pub cudnnStatus_t);
+
+impl cudnnStatus_t {
+    /// Transforms into a [Result] of [CudnnError]
+    pub fn result(self) -> Result<(), CudnnError> {
+        match self {
+            cudnnStatus_t::CUDNN_STATUS_SUCCESS => Ok(()),
+            _ => Err(CudnnError(self)),
+        }
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::fmt::Display for CudnnError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{self:?}")
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for CudnnError {}
+
+// No sync because of https://docs.nvidia.com/deeplearning/cudnn/developer-guide/index.html#thread-safety
+pub struct CudnnHandle(pub(crate) cudnnHandle_t);
```

Comment:
This needs to keep a reference to Arc<CudaDevice> to make sure this doesn't outlive device:
```suggestion
pub struct CudnnHandle {
    pub(crate) handle: cudnnHandle_t,
    pub(crate) device: Arc<CudaDevice>,
}
```

And then `fn create` should take an `Arc<CudaDevice>`.

Did a similar thing with blas if you want an example

Reply:
ok, didn't think of that 😅 

by the way, the test "launch with 16 bit" fails for me with an buffer overflow; are you having the same issue?

Reply:
I haven't, I can check again

Reply:
ok thanks